### PR TITLE
Adding support for the Adafruit QT Py ESP32-S3

### PR DESCRIPTION
**Pull request by raveious**
This adds support for the Adafruit QT Py ESP32-S3 board for both the "non-psram" variant as well as the "with psram" variant. The QT Py ecosystem is designed to be compatible with Seeedstudio's Xiao ecosystem, while bringing integration with the rest of Adafruit's STEMMA QT system.

### CODE REVIEW

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_common.dtsi
Comment (nordicjm) [inline boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_common.dtsi:81]: too much indent

File: boards/adafruit/qt_py_esp32s3/board.yml
Comment (nordicjm) [inline boards/adafruit/qt_py_esp32s3/board.yml:11]: if this comes in 2 designs then it should be using hardware versions not variants unless you mean there is only 1 board and you can pick if you use the psram or not

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/board.yml:11]: Adafruit offers 2 boards with the same name, with the only difference being that one has "with PSRAM" in the name. The best I can tell, they use the same schematic and use a BOM swap to have the different SOCs placed in manufacturing.

Both products are still in active production:
- [Adafruit QT Py ESP32-S3 WiFi Dev Board with STEMMA QT - 8 MB Flash / No PSRAM](https://www.adafruit.com/product/5426)
- [Adafruit QT Py S3 with 2MB PSRAM WiFi Dev Board with STEMMA QT](https://www.adafruit.com/product/5700)

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_common.dtsi
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_common.dtsi:81]: globally or specifically right here?

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_common.dtsi:81]: oh, I see it now. this whole section.

File: boards/adafruit/qt_py_esp32s3/board.yml
Comment (nordicjm) [reply boards/adafruit/qt_py_esp32s3/board.yml:11]: Use board revisions: https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#multiple-board-revisions

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/board.yml:11]: Except the boards aren't revisions of each other, they are distinct products and don't fall into a semantic versioning scheme. Unless you're trying to suggest that using the amount of flash a board has should be using the revision scheme?

Comment (nordicjm) [reply boards/adafruit/qt_py_esp32s3/board.yml:11]: If they're distinct products then why would they be variants?

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/board.yml:11]: The only difference between the two variants is that one includes in their device tree

```
#include <espressif/esp32s3/esp32s3_wroom_n8.dtsi>
```

while the other includes

```
#include <espressif/esp32s3/esp32s3_wroom_n4r2.dtsi>
```

The other way to accomplish this would be to have 2 complete board directories with near identical contents (including docs), and that seems a bit overkill for what is a 1 line change. I felt that the most effective way to support both versions of the board is to use the variant system.

It gets even more confusing with the physical boards too, as the only way to visually tell the boards apart is to look at the part number on the SOC itself.

**Comment by simonguinot** [comment]
Passing this PR to the Espressif maintainers.

File: boards/adafruit/qt_py_esp32s3/board.yml
Comment (sylvioalves) [reply boards/adafruit/qt_py_esp32s3/board.yml]: Having 2 different boards just because SPIRAM sounds odd, I agree about that. 
At the same time, It sounds those boards have no "revision" number between those 2 parts.

The 3rd option and simpler approach would be to only include the board without PSRAM and let application handling additional .conf/.overlay files to include PSRAM entries. You can add this info into documentation describing how to do so..

File: boards/adafruit/qt_py_esp32s3/doc/index.rst
Comment (sylvioalves) [inline boards/adafruit/qt_py_esp32s3/doc/index.rst:11]: I think you can remove this figure entry from here. It shall be automatically added. @kartben

File: boards/adafruit/qt_py_esp32s3/doc/index.rst
Comment (sylvioalves) [inline boards/adafruit/qt_py_esp32s3/doc/index.rst:73]: Sounds Wi-Fi and BLE are missing.

File: boards/adafruit/qt_py_esp32s3/doc/index.rst
Comment (sylvioalves) [inline boards/adafruit/qt_py_esp32s3/doc/index.rst:190]: Can you fix this?
`XIAO ESP32S3 Sense`

File: boards/adafruit/qt_py_esp32s3/doc/index.rst
Comment (sylvioalves) [inline boards/adafruit/qt_py_esp32s3/doc/index.rst:26]: typo: "`is` also has a RGB NeoPixel for", should be `it`.

**Comment by sylvioalves** [comment]
@raveious Thanks for the update. I have no additional comments.

File: boards/adafruit/qt_py_esp32s3/doc/index.rst
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/doc/index.rst:190]: Crap..... I thought I got them all..... Now you know where I copied this from

**Review by nordicjm — COMMENTED** [review]
this should fix it

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts:5]: For my own understanding, you are explicitly deleting this node to make sure it gets replaced by the one from espressif/esp32s3/esp32s3_wroom_n4r2.dtsi? Do you need to do this with normal dts inclusions?

Comment (nordicjm) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts:5]: By having a board without a board revision, it will not have an overlay file and will only load the dts configuration from the .dts file, which is a problem as you need to have a different file included for each revision, so the workaround is allow the original be included, then delete what the original sets, and include the one that is wanted

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: I also probably want to delete the `psram0` node as well.

Comment (nordicjm) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: > The overlay probably shouldn't be including the 'dtsi' files I had, right?

Correct, they will get it from the base .dts file

> I also probably want to delete the psram0 node as well.

I don't see that, the .overlay file is for the psnode one so it should have it defined

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: > The overlay probably shouldn't be including the 'dtsi' files I had, right?

Nope, they are very much required. Doesn't seem to build without them.

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: > > The overlay probably shouldn't be including the 'dtsi' files I had, right?
> 
> Correct, they will get it from the base .dts file

When removing that include, I get the following device tree error:

```
devicetree error: /chosen: undefined node label 'boot_partition'
```

CMake is reporting that BOARD.dts is adafruit_qt_py_esp32s3_procpu.dts (which is correct).
It then shows the devicetree overlay as adafruit_qt_py_esp32s3_procpu_psram.overlay and app.overlay from mcuboot.

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: I was able to get past the boot partition issue by re-including "espressif/partitions_0x0_amp.dtsi" in the overlay, but it then wasn't able to find a console driver as it believed that all of the espressif,esp32-uart nodes were missing.

I'm starting to think there might be a limitation of how dtc and the scripts that run it are used, and we're hitting that limitation by not re-including the common dtsi file.

What I have pushed at least builds, but not as confident that it is as correct as the original "variant" implementation I originally had.

Comment (marekmatej) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: @raveious try to move the `#include <espressif/partitions_0x0_amp.dtsi>` after other includes as it referencing the `flash0` node.

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: > @raveious try to move the `#include <espressif/partitions_0x0_amp.dtsi>` after other includes as it referencing the `flash0` node.

I tried putting it at the end of the includes in the device tree with no effect. When putting at the end of the includes for the overlay, it makes it a bit further but fails with the following series of errors:
```
warning: UART_CONSOLE (defined at drivers/console/Kconfig:42) was assigned the value 'y' but got the
value 'n'.
<snip>
CMake Warning at <snip>/zephyr/CMakeLists.txt:957 (message):
  No SOURCES given to Zephyr library: drivers__console

  Excluding target from build.


CMake Warning at <snip>/zephyr/CMakeLists.txt:957 (message):
  No SOURCES given to Zephyr library: drivers__serial

  Excluding target from build.


CMake Warning at <snip>/zephyr/CMakeLists.txt:957 (message):
  No SOURCES given to Zephyr library: drivers__spi

  Excluding target from build.
```
Ultimately failing with a undefined reference to `bootloader_console_init`. This seems to be because `DT_HAS_ESPRESSIF_ESP32_UART_ENABLED` is missing from the generated device tree header. That define gets setup when it finds `espressif,esp32-uart` which is defined in `dts/xtensa/espressif/esp32s3/esp32s3_common.dtsi` and enabled in `adafruit_qt_py_esp32s3_procpu.dts` (or *_common.dtsi if you are looking at this PR). However, it seems to be ignored even though I see if being sucked into the build.

I've pushed the commits to [here](https://github.com/raveious/zephyr/commit/291d87a7d0bc8567d21894fdf33d76e716bcdba4) and I've been building with one of the following commands:
```
west build -b adafruit_qt_py_esp32s3@psram/esp32s3/procpu --sysbuild samples/drivers/led/led_strip
west build -b adafruit_qt_py_esp32s3@psram/esp32s3/procpu --sysbuild samples/hello_world
```

Comment (nordicjm) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: Apply this patch, should be fine after: https://pastebin.com/Z113TaMk

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: > Apply this patch, should be fine after: https://pastebin.com/Z113TaMk

That patch does fix the build. It does seem odd that this has to be copied from the SOC dtsi files though. Won't this cause maintenance issues in the future by having snippets like this all over the place?

File: boards/adafruit/qt_py_esp32s3/board.yml
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/board.yml:12]: I'm not sure if the empty string is actually going to work as twister found a cmake build failure:

```
Loading Zephyr default modules (Zephyr base).
-- Application: <snip>/zephyr/tests/application_development/gen_inc_file
-- CMake version: 3.30.3
-- Found Python3: /Projects/zephyr-venv/bin/python (found suitable version "3.12.6", minimum required is "3.10") found components: Interpreter
-- Cache files will be written to: <snip>
-- Zephyr version: 4.0.99 (<snip>/zephyr)
-- Found west (found suitable version "1.3.0", minimum required is "0.14.0")
CMake Error at <snip>/zephyr/cmake/modules/boards.cmake:73 (message):
  Invalid revision / qualifiers format for BOARD
  (adafruit_qt_py_esp32s3@/esp32s3/procpu).  Valid format is:
  <board>@<revision>/<qualifiers>
Call Stack (most recent call first):
  <snip>/zephyr/cmake/modules/boards.cmake:85 (parse_board_components)
  <snip>/zephyr/cmake/modules/zephyr_default.cmake:133 (include)
  <snip>/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  <snip>/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
```

Comment (nordicjm) [reply boards/adafruit/qt_py_esp32s3/board.yml:12]: the non-PSRAM twister .yaml files should not have an @ in, only the PSRAM ones

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts
Comment (nordicjm) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: So essentially this is a hack, board revisions are used for textual/numerical board revisions, having a board revision where one is empty is a special case for this board because of how the manufacturer has decided to produce a badly labelled board

File: boards/adafruit/qt_py_esp32s3/board.yml
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/board.yml:12]: > the non-PSRAM twister .yaml files should not have an @ in, only the PSRAM ones

Yup, thats why I don't have a @ in there. I wonder if twister is looking at board.yml for revisions and assuming everything gets a leading @ when generating tests?

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.dts]: > So essentially this is a hack, board revisions are used for textual/numerical board revisions, having a board revision where one is empty is a special case for this board because of how the manufacturer has decided to produce a badly labelled board

I completely agree that every board should have a revision on it. I guess my question is, knowing how much of a hack this feels, was this better or worse than the variants approach?

File: boards/adafruit/qt_py_esp32s3/doc/index.rst
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/doc/index.rst]: Looking at some of the other examples, I think you are correct. I have removed it.

Comment (kartben) [reply boards/adafruit/qt_py_esp32s3/doc/index.rst]: You may actually see the docs for your PR here :)

https://builds.zephyrproject.io/zephyr/pr/81857/docs/boards/adafruit/qt_py_esp32s3/doc/index.html
https://builds.zephyrproject.io/zephyr/pr/81857/docs/boards/index.html#name=Qt

**Comment by raveious** [comment]
> @sylvioalves @nordicjm Based on feedback from discord, and from looking around in the twister code (specifically, this [line](https://github.com/zephyrproject-rtos/zephyr/blob/89d4aa85592dac196e25a157566cd6f67877dd1d/scripts/pylib/twister/twisterlib/testplan.py#L509%5D)); I've come to the conclusion that having an empty string board revision name is not a good idea, nor are revisions with an empty string name supported at this time. I would like to revert the change back to the earlier implementation that used board variants as that still seems to be the correct way of implementing this.

Alternatively, I'd also consider splitting everything into their own board folders. That does come with the issue of having 2 doc pages.

**Comment by nordicjm** [comment]
Board revisions is the correct way to handle this, not board variants. It can use a board revision with a string for the default board in that case

**Comment by nordicjm** [comment]
@nashif @PerMac seems like a bug in twister here, it's providing an empty board revision when it should just not provide one, can demonstrate the issue with this:
```
variants:
  adafruit_qt_py_esp32s3@/esp32s3/procpu:
    twister: false
  adafruit_qt_py_esp32s3@/esp32s3/appcpu:
    twister: false
```
And adjusting a sample to have:
```
  integration_platforms:
    - adafruit_qt_py_esp32s3@/esp32s3/procpu
    - adafruit_qt_py_esp32s3@/esp32s3/appcpu
    - adafruit_qt_py_esp32s3/esp32s3/procpu
    - adafruit_qt_py_esp32s3/esp32s3/appcpu
```
With both board targets uncommented:
```
ERROR   - Platform adafruit_qt_py_esp32s3@/esp32s3/procpu not found in the list of platforms
ERROR   - Platform adafruit_qt_py_esp32s3@/esp32s3/appcpu not found in the list of platforms
ERROR   - Platform adafruit_qt_py_esp32s3/esp32s3/procpu not found in the list of platforms
ERROR   - Platform adafruit_qt_py_esp32s3/esp32s3/appcpu not found in the list of platforms
```
With the top board target commented and button uncommented:
```
ERROR   - Platform adafruit_qt_py_esp32s3@/esp32s3/appcpu not found in the list of platforms
ERROR   - Platform adafruit_qt_py_esp32s3/esp32s3/appcpu not found in the list of platforms
```

**Comment by nordicjm** [comment]
@raveious should work once above fix is merged

**Comment by nordicjm** [comment]
@raveious try rebasing on main, the original board should then work

**Comment by raveious** [comment]
> @raveious try rebasing on main, the original board should then work

Rebase completed.

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml
Comment (sylvioalves) [inline boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml:18]: Would you please remove the below? We have started removing this entries from all boards (another PR). Would be good not to keep adding this anymore.
```c
testing:
  ignore_tags:
    - net
    - bluetooth
```

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.overlay
Comment (sylvioalves) [inline boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.overlay:11]: @marekmatej PTAL.

**Comment by sylvioalves** [comment]
LGTM, just one last request.

**Review by wmrsouza — DISMISSED** [review]

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml:18]: Should I also be removing the `net` and `bluetooth` from the `ignore_tags` section of the appcpu.yaml file?

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml:18]: is it PR #80785 ?

Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml]: I took the lines out of the procpu.yaml files.

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_appcpu.dts
Comment (wmrsouza) [inline boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_appcpu.dts:16]: Please, change to sram1 to take in account #80340

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml
Comment (wmrsouza) [inline boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml:17]: I think you missed i2s

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.yaml
Comment (wmrsouza) [inline boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.yaml:17]: i2s missed here also

File: boards/adafruit/qt_py_esp32s3/doc/index.rst
Comment (wmrsouza) [inline boards/adafruit/qt_py_esp32s3/doc/index.rst:43]: i2s missed here also

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.dts
Comment (wmrsouza) [inline boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.dts:95]: only for the records: another option would be have led_strip under i2s bus

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.dts
Comment (wmrsouza) [inline boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.dts:22]: Please change to sram1 to be compliant with #80340

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.yaml
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu_psram.yaml:17]: This board is almost completely copied from seeed_xiao_esp32s3, and its missing from there too. I'll add it to this board, but we should add it to the Xiao board if that truely is the case.

File: boards/adafruit/qt_py_esp32s3/doc/index.rst
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/doc/index.rst:43]: This board is almost completely copied from seeed_xiao_esp32s3, and its missing from there too. I'll add it to this board, but we should add it to the Xiao board if that truely is the case.

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.yaml:17]: This board is almost completely copied from seeed_xiao_esp32s3, and its missing from there too. I'll add it to this board, but we should add it to the Xiao board if that truely is the case.

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.dts
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.dts:95]: Personally, I think that misleading as its using the SPI driver for the LED.

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_appcpu.dts
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_appcpu.dts]: Addressed.

File: boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.dts
Comment (raveious) [reply boards/adafruit/qt_py_esp32s3/adafruit_qt_py_esp32s3_procpu.dts]: addressed in second push.

**Review by wmrsouza — APPROVED** [review]

**Comment by nashif** [comment]
> Board revisions is the correct way to handle this, not board variants. It can use a board revision with a string for the default board in that case


@nordicjm  wondering if this is correct usage of revisions, to me this looks like a board with multiple configurations rather than different revisions. IMO this is misuse of the custom revision feature.

This board comes with no psram/wih psram (https://www.adafruit.com/product/5426 vs https://www.adafruit.com/product/5700), it is not a board that was updated and has different versions.

I have seen this being done for many other boards. Why is variant not suitable to handle this case?

**Comment by raveious** [comment]
> > Board revisions is the correct way to handle this, not board variants. It can use a board revision with a string for the default board in that case
> 
> @nordicjm wondering if this is correct usage of revisions, to me this looks like a board with multiple configurations rather than different revisions. IMO this is misuse of the custom revision feature.
> 
> This board comes with no psram/wih psram (https://www.adafruit.com/product/5426 vs https://www.adafruit.com/product/5700), it is not a board that was updated and has different versions.
> 
> I have seen this being done for many other boards. Why is variant not suitable to handle this case?

Funny enough, the original implementation used board variants. I was asked to change it to use board revision instead.

**Comment by nordicjm** [comment]
> > Board revisions is the correct way to handle this, not board variants. It can use a board revision with a string for the default board in that case
> 
> @nordicjm wondering if this is correct usage of revisions, to me this looks like a board with multiple configurations rather than different revisions. IMO this is misuse of the custom revision feature.
> 
> This board comes with no psram/wih psram (https://www.adafruit.com/product/5426 vs https://www.adafruit.com/product/5700), it is not a board that was updated and has different versions.
> 
> I have seen this being done for many other boards. Why is variant not suitable to handle this case?

Variant is, as per the terminology:

> In the context of [board qualifiers](https://docs.zephyrproject.org/latest/glossary.html#term-board-qualifiers), a variant designates a particular type or configuration of a build for a combination of [SoC](https://docs.zephyrproject.org/latest/glossary.html#term-SoC) and [CPU cluster](https://docs.zephyrproject.org/latest/glossary.html#term-CPU-cluster). Common uses of the variant concept include introducing both secure and non-secure builds for platforms with Trusted Execution Environment support, or selecting the type of RAM used in a build.

So this means you have one board and you can run a number of different configurations on it, the RAM part applies to e.g. some NXP boards where you could be running from SRAM, DRAM, TCM, etc. but they all work on the same board. This board here is actually 2 different boards but the manufacturer has called them the exact same thing. You cannot use the psram one on the non-psram board because it does not have psram, therefore it is not a variant. See terminology for board revision:

> An optional version string that identifies a particular revision of a hardware system. This is useful to avoid duplication of board files whenever small changes are introduced to a hardware system. See [Multiple board revisions](https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#porting-board-revisions) and [Building for a board revision](https://docs.zephyrproject.org/latest/develop/application/index.html#application-board-version) for more information.

The board files are mostly the same which is why there is only one set for the base defconfig and dts files, the only difference is one has the psram memory layout and the other does not, so there is an overlay for the psram version to enable that. Sure the intended use of board revision is for when the version of the actual hardware is updated but given that this is 2 different boards with the exact same name that are incompatible due to different SoC part numbers I don't a way of having 2 different boards

**Comment by nashif** [comment]
> You cannot use the psram one on the non-psram board because it does not have psram, therefore it is not a variant. See terminology for board revision:

If it is a different board, it should not be a revision, it should be added as a new  board entry in board.yml.

Variants is used in the tree for the same board with different HW features, for example right now we have rpi_pico with a /w variant, those are 2 different boards using the same name with different features. Other example exist, i.e. `XIAO BLE (Sense)` with a `/sense` variants, not sure how that fits. 

We could also expand the definition of variants to suppor those cases of same board with different configurations and avoid creating new boards or misuing revisions.

**Comment by nordicjm** [comment]
> If it is a different board, it should not be a revision, it should be added as a new board entry in board.yml.

The problem is that it's the same board, it's just a different SoC fitted to the board, they are both "qt py esp32-s32" so you can't use a board for that (unless you suffix one with _psram which is going to be similar to this)

> Variants is used in the tree for the same board with different HW features, for example right now we have rpi_pico with a /w variant, those are 2 different boards using the same name with different features. Other example exist, i.e. XIAO BLE (Sense) with a /sense variants, not sure how that fits.

Mistakes were made in hwmv2 porting because people like me with no knowledge of a lot of boards rushed to convert 600+ boards into hwmv2 format, variants should not have been used for that. Likewise when you reworked something in twister recently affecting the stm32 min dev board which I then noticed had a default revision in the hwmv2 conversion, which it should not

> We could also expand the definition of variants to support those cases of same board with different configurations and avoid creating new boards or misuing revisions.

But this is a revision or a separate board and not a variant, if a vendor releases a v1.0 board then a v1.5 board which might change the CPU, that's still a revision of the board, variant is for things like smp or non-smp e.g. on qemu boards or the non-secure core on trustzone supported CPUs or selecting which active configuration is used, I don't see it as useful to make variant and revision into the same thing, there should be a clear distinction between them. A board might ship with a wrong connection to a sensor then an update might fix it, there is no way you can get the sensor on the original board working whereas a non-secure core will always work on a specific board, variant should be virtual and revision should be physical

**Review by nordicjm — APPROVED** [review]

**Comment by kartben** [comment]
ping @sylvioalves

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**
